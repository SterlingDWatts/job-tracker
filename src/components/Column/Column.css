.Column {
  font-size: 12px;
  font-weight: 400;
  background-color: #f4f5f7;
  color: rgb(23, 43, 77);
  min-width: 200px;
}

.Column .card-container {
  padding: 1px 3px 0px;
}

.Column + .Column {
  margin-left: 10px;
}
