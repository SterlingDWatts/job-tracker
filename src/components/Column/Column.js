import React from "react";
import Card from "../Card/Card";
import "./Column.css";

export default function Column(props) {
  let applications = props.applications;
  if (applications) {
    applications = applications.filter(
      (application) =>
        application.column[application.column.length - 1].id === props.id
    );
  }
  if (applications) {
    applications = applications.map((application) => (
      <Card application={application} key={application.id} />
    ));
  }
  return (
    <div className="Column">
      <h2>{props.name}</h2>
      <div className="card-container">{applications}</div>
    </div>
  );
}
