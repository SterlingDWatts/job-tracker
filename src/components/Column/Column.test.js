import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import Column from "./Column";

describe("Column Comopnent", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Column />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = shallow(<Column id={0} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
