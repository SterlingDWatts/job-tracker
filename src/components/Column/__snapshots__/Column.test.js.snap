// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Comopnent renders the UI as expected 1`] = `
<div
  className="Column"
>
  <div
    className="card-container"
  />
</div>
`;
